### PR TITLE
Update deployment support for Azure China

### LINKED ISSUE #259: [Hubs] Add support for the China cloud
## ⚠️ Problem
Microsoft Cost Management added support for Microsoft Customer Agreement accounts in China in July. We need to confirm that FinOps hubs is deployable in that cloud.

## 🛠️ Solution
Follow the documentation to deploy FinOps hubs in China and identify any issues with the documentation or deployment that cause it to not work.

## 🙋‍♀️ Ask for the community

We could use your help:
1. Please vote this issue up (👍) to prioritize it.
2. Leave comments to help us solidify the vision.

### PR DESCRIPTION
**Pull request by yuanzhang9**
## 🛠️ Description
Add support for Azure China in Finops hubs.

Fixes #259 

## 🔬 How has this been tested?

- [ ] 🫰 PS -WhatIf / az validate
- [ ] 👍 Manually deployed + verified
- [ ] 💪 Unit tests

### CODE REVIEW

**Comment by flanakin** [comment]
@yuanzhang9 This change looks good. Can you also add the bicep changes needed to support China?

**Review by flanakin — CHANGES_REQUESTED** [review]
@yuanzhang9 This change looks good. Can you also add the bicep changes needed to support China?

**Comment by yuanzhang9** [comment]
Update to support China region for both Bicep and ARM

**Comment by yuanzhang9** [comment]
@flanakin new commits to the same PR are Bicep files as well as the read me file.
Please help review.

File: src/templates/finops-hub/modules/dataFactory.bicep [hunk 1]
```diff
@@ -102,7 +102,7 @@ resource identityRoleAssignments 'Microsoft.Authorization/roleAssignments@2022-0
 // Stop hub triggers if they're already running
 resource stopHubTriggers 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
   name: '${dataFactoryName}_stopHubTriggers'
-  location: location
+  location: startsWith(location, 'china') ? 'chinaeast2' : location // Pin to the only region(chinaeast2) that supports deployment scripts in China
```
Comment (flanakin) [inline src/templates/finops-hub/modules/dataFactory.bicep:105]: nit: Can we put this on the line above for readability?
```suggestion
  // chinaeast2 is the only region in China that supports deployment scripts
  location: startsWith(location, 'china') ? 'chinaeast2' : location
```

File: src/templates/finops-hub/modules/dataFactory.bicep [hunk 2]
```diff
@@ -677,7 +677,7 @@ resource pipeline_transformExport 'Microsoft.DataFactory/factories/pipelines@201
 // Start hub triggers
 resource startHubTriggers 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
   name: '${dataFactoryName}_startHubTriggers'
-  location: location
+  location: startsWith(location, 'china') ? 'chinaeast2' : location // Pin to the only region(chinaeast2) that supports deployment scripts in China
```
Comment (flanakin) [inline src/templates/finops-hub/modules/dataFactory.bicep:680]:
```suggestion
  // chinaeast2 is the only region in China that supports deployment scripts
  location: startsWith(location, 'china') ? 'chinaeast2' : location
```

File: src/templates/finops-hub/modules/keyVault.bicep [hunk 3]
```diff
@@ -1,4 +1,3 @@
-// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
```
Comment (flanakin) [inline src/templates/finops-hub/modules/keyVault.bicep:1]:
```suggestion
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
```

File: src/templates/finops-hub/modules/keyVault.bicep [hunk 4]
```diff
@@ -65,7 +64,7 @@ resource keyVault 'Microsoft.KeyVault/vaults@2022-11-01' = {
     tenantId: subscription().tenantId
     accessPolicies: formattedAccessPolicies
     sku: {
-      name: sku
+      name: startsWith(location, 'china') ? 'standard' : sku // Pin to "standard" keyvault which is the only supported sku in China
```
Comment (flanakin) [inline src/templates/finops-hub/modules/keyVault.bicep:67]:
```suggestion
      // chinaeast2 is the only region in China that supports deployment scripts
      name: startsWith(location, 'china') ? 'standard' : sku
```

File: src/templates/finops-hub/modules/storage.bicep [hunk 5]
```diff
@@ -122,7 +122,7 @@ resource identityRoleAssignments 'Microsoft.Authorization/roleAssignments@2022-0
 resource uploadSettings 'Microsoft.Resources/deploymentScripts@2020-10-01' = {
   name: 'uploadSettings'
   kind: 'AzurePowerShell'
-  location: location
+  location: startsWith(location, 'china') ? 'chinaeast2' : location // Pin to the only region(chinaeast2) that supports deployment scripts in China
```
Comment (flanakin) [inline src/templates/finops-hub/modules/storage.bicep:125]:
```suggestion
  // chinaeast2 is the only region in China that supports deployment scripts
  location: startsWith(location, 'china') ? 'chinaeast2' : location
```

**Review by flanakin — APPROVED** [review]

**Review by flanakin — APPROVED** [review]

**Pull request merged**